### ISSUE TITLE
Last Update Error: Setting unknown property: kartik\date\DatePicker::aria-required

### ISSUE THREAD

**Issue by serhatozles** [yiisoft/yii2#13140]
I'm using this code;
```php
<?= $form->field($model, 'finish_date')->widget(DateControl::classname(), [
                        'type' => DateControl::FORMAT_DATE,
                        'displayFormat' => 'dd/MM/yyyy',
                        'options' => [
                            'pluginOptions' => [
                                'id' => 'finish_date',
                                'autoclose' => true,
                                'todayHighlight' => true,
                                'todayBtn' => true,
                                'startDate' => "d",
                            ],
                            'layout' => '{picker}{input}',
                        ],
                    ]) ?>
```
And This function is giving error from "ActiveField";
```php
    protected function addAriaAttributes(&$options)
    {
        if ($this->addAriaAttributes) {
            if (!isset($options['aria-required']) && $this->model->isAttributeRequired($this->attribute)) {
                $options['aria-required'] =  'true';
            }
            if (!isset($options['aria-invalid'])) {
                if ($this->model->hasErrors($this->attribute)) {
                    $options['aria-invalid'] = 'true';
                }
            }
        }
    }
```

If i put old version to ActiveField is working....
Can you help me to solve this?

**Comment by dynasource** [comment]
the feature aria attributes has just been added to the framework. Can you post the error log?

**Comment by samdark** [comment]
Error message along with stacktrace would be really helpful.

**Comment by serhatozles** [comment]
here is it..
`exception 'yii\base\UnknownPropertyException' with message 'Setting unknown property: kartik\date\DatePicker::aria-required' in C:\Winginx\home\kib.loc\public_html\yii2-client\vendor\yiisoft\yii2\base\Component.php:201
Stack trace:
#0 C:\Winginx\home\kib.loc\public_html\yii2-client\vendor\yiisoft\yii2\BaseYii.php(521): yii\base\Component->__set('aria-required', 'true')
#1 C:\Winginx\home\kib.loc\public_html\yii2-client\vendor\yiisoft\yii2\base\Object.php(105): yii\BaseYii::configure(Object(kartik\date\DatePicker), Array)
#2 [internal function]: yii\base\Object->__construct(Array)
#3 C:\Winginx\home\kib.loc\public_html\yii2-client\vendor\yiisoft\yii2\di\Container.php(381): ReflectionClass->newInstanceArgs(Array)
#4 C:\Winginx\home\kib.loc\public_html\yii2-client\vendor\yiisoft\yii2\di\Container.php(156): yii\di\Container->build('kartik\date\Dat...', Array, Array)
#5 C:\Winginx\home\kib.loc\public_html\yii2-client\vendor\yiisoft\yii2\BaseYii.php(344): yii\di\Container->get('kartik\date\Dat...', Array, Array)
#6 C:\Winginx\home\kib.loc\public_html\yii2-client\vendor\yiisoft\yii2\base\Widget.php(103): yii\BaseYii::createObject(Array)
#7 C:\Winginx\home\kib.loc\public_html\yii2-client\vendor\kartik-v\yii2-datecontrol\DateControl.php(297): yii\base\Widget::widget(Array)
#8 C:\Winginx\home\kib.loc\public_html\yii2-client\vendor\kartik-v\yii2-datecontrol\DateControl.php(249): kartik\datecontrol\DateControl->getDisplayInput()
#9 C:\Winginx\home\kib.loc\public_html\yii2-client\vendor\yiisoft\yii2\base\Widget.php(104): kartik\datecontrol\DateControl->run()
#10 C:\Winginx\home\kib.loc\public_html\yii2-client\vendor\yiisoft\yii2\widgets\ActiveField.php(720): yii\base\Widget::widget(Array)
#11 C:\Winginx\home\kib.loc\public_html\yii2-client\vendor\kartik-v\yii2-widget-activeform\ActiveField.php(710): yii\widgets\ActiveField->widget('kartik\datecont...', Array)
#12 C:\Winginx\home\kib.loc\public_html\yii2-client\backend\modules\pricesale\views\period\_form.php(38): kartik\form\ActiveField->widget('kartik\datecont...', Array)
#13 C:\Winginx\home\kib.loc\public_html\yii2-client\vendor\yiisoft\yii2\base\View.php(328): require('C:\Winginx\home...')
#14 C:\Winginx\home\kib.loc\public_html\yii2-client\vendor\yiisoft\yii2\base\View.php(250): yii\base\View->renderPhpFile('C:\Winginx\home...', Array)
#15 C:\Winginx\home\kib.loc\public_html\yii2-client\vendor\yiisoft\yii2\base\View.php(152): yii\base\View->renderFile('C:\Winginx\home...', Array, NULL)
#16 C:\Winginx\home\kib.loc\public_html\yii2-client\common\modules\tools\view\Tools.php(35): yii\base\View->render('_form', Array, NULL)
#17 C:\Winginx\home\kib.loc\public_html\yii2-client\backend\modules\pricesale\views\period\update.php(23): common\modules\tools\view\Tools->render('_form', Array)
#18 C:\Winginx\home\kib.loc\public_html\yii2-client\vendor\yiisoft\yii2\base\View.php(328): require('C:\Winginx\home...')
#19 C:\Winginx\home\kib.loc\public_html\yii2-client\vendor\yiisoft\yii2\base\View.php(250): yii\base\View->renderPhpFile('C:\Winginx\home...', Array)
#20 C:\Winginx\home\kib.loc\public_html\yii2-client\vendor\yiisoft\yii2\base\View.php(152): yii\base\View->renderFile('C:\Winginx\home...', Array, Object(backend\modules\pricesale\controllers\PeriodController))
#21 C:\Winginx\home\kib.loc\public_html\yii2-client\common\modules\tools\view\Tools.php(35): yii\base\View->render('update', Array, Object(backend\modules\pricesale\controllers\PeriodController))
#22 C:\Winginx\home\kib.loc\public_html\yii2-client\vendor\yiisoft\yii2\base\Controller.php(379): common\modules\tools\view\Tools->render('update', Array, Object(backend\modules\pricesale\controllers\PeriodController))
#23 C:\Winginx\home\kib.loc\public_html\yii2-client\backend\components\BetaBaseController.php(31): yii\base\Controller->render('update', Array)
#24 C:\Winginx\home\kib.loc\public_html\yii2-client\backend\modules\pricesale\controllers\PeriodController.php(122): backend\components\BetaBaseController->render('update', Array)
#25 [internal function]: backend\modules\pricesale\controllers\PeriodController->actionUpdate('53', '24')
#26 C:\Winginx\home\kib.loc\public_html\yii2-client\vendor\yiisoft\yii2\base\InlineAction.php(57): call_user_func_array(Array, Array)
#27 C:\Winginx\home\kib.loc\public_html\yii2-client\vendor\yiisoft\yii2\base\Controller.php(156): yii\base\InlineAction->runWithParams(Array)
#28 C:\Winginx\home\kib.loc\public_html\yii2-client\vendor\yiisoft\yii2\base\Module.php(522): yii\base\Controller->runAction('update', Array)
#29 C:\Winginx\home\kib.loc\public_html\yii2-client\vendor\yiisoft\yii2\web\Application.php(102): yii\base\Module->runAction('pricesale/perio...', Array)
#30 C:\Winginx\home\kib.loc\public_html\yii2-client\vendor\yiisoft\yii2\base\Application.php(379): yii\web\Application->handleRequest(Object(yii\web\Request))
#31 C:\Winginx\home\kib.loc\public_html\web-com2\index.php(18): yii\base\Application->run()
#32 {main}`

**Comment by xReprisal** [comment]
Aria attributes are passed in options array to the widget.
Problem arises when widget is using those options to instantiate itself:
`DateControl` line 297
`return $class::widget($this->options);`

Quick fix would be to disable addAriaAttributes for this field: 
`$form->field($model, 'finish_date', ['addAriaAttributes' => false])`

I'm sure there was a reason behind implementing `DateControl` widget in such way. Nevertheless, I think widget  initialization should look more like below and not use options as variables.
```
<?= $form->field($model, 'finish_date')->widget(DateControl::classname(), [
                        'type' => DateControl::FORMAT_DATE,
                        'displayFormat' => 'dd/MM/yyyy',
                        'layout' => '{picker}{input}',
                        'options' => [
                            'id' => 'finish_date',
                            'autoclose' => true,
                            'todayHighlight' => true,
                            'todayBtn' => true,
                            'startDate' => "d",
                        ],
                    ]) ?>
```

**Comment by cebe** [comment]
> Problem arises when widget is using those options to instantiate itself:

options are clearly documented to be used as HTML attributes and not to configure another object...

> http://www.yiiframework.com/doc-2.0/yii-widgets-inputwidget.html#$options-detail
> The HTML attributes for the input tag.
> See also yii\helpers\Html::renderTagAttributes() for details on how attributes are being rendered.

This is wrong usage of the property by the DateControl widget in kartik-v/yii2-datecontrol package.
Seems it has a double meaning, once used as properties of a widget and once as HTML attributes.
https://github.com/kartik-v/yii2-datecontrol/blob/31f05b03d3e28f45e2b16ffd9e01f7d8e29da5a2/DateControl.php#L110-L114

There is nothing to fix in yii here, its a bug in the widget.

**Comment by cebe** [comment]
You can work around it by disabling `ActiveField::$addAriaAttributes`.

**Comment by kartik-v** [comment]
This is been fixed in v1.9.5 release of the widget [via this change](https://github.com/kartik-v/yii2-datecontrol/commit/497b7bb438864843b450aa9ab1df20bed096b5a1). This will be a BC breaking change and you can set the new property `DateControl::widgetOptions` now.
